### PR TITLE
docs(Data Privacy): Create data privacy docs

### PR DESCRIPTION
**Pull request by justinanastos**
I agonized over this for a while and then remembered this quote:

>Release when something is useful, not when it's perfect.

Please give these a proof-read!

---

Addresses most of #145 , with the exceptions of:

- GDPR #165 
- Alerts channel configuration #165

### CODE REVIEW

File: source/data-privacy.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+### Engine Proxy
+
+While the Engine proxy sees your client request data and service response data, it only collects and forwards data that goes into the reports you see in the Engine dashboards. All information that is sent from your on-prem proxy to the out-of-band Engine cloud service is configurable and can be turned off through config options. Data is aggregated and sent approximately every 5 seconds.
+
+Let's walk through Engine's default behavior for reporting on fields in a typical GraphQL response:
+
+```
+// GraphQL Response
+{
+  "data": { ... },          // Engine does not read this field, but might cache the response.
```
Comment (stubailo) [inline source/data-privacy.md]: > Engine does not read this field, but might cache the response

I'd maybe clarify and say "contents of this field are never sent to the engine cloud service" or similar. Right now it feels like I have to read the clarification below to get the main info.

File: source/data-privacy.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+<h3 id="disable-reporting" title="Disable Reporting">Disable Reporting</h3>
+
+We've added the option to disable reporting of stats and traces to Apollo servers so that integration tests can run without polluting production data.
+
+To disable all reporting, use the [`disabled`](./proxy-config.html#Reporting) option.
+
+<h3 id="http-headers">Authorization & Cookie HTTP Headers</h3>
+
+We'll **never** collect your application's `Authorization` headers, `Cookie`, or `Set-Cookie` headers.
```
Comment (stubailo) [inline source/data-privacy.md]: This raises the question - what do we do with other headers? Do we collect them all?

File: source/data-privacy.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+} 
+```
+
+#### `response.data`
+
+The Engine proxy will never send the contents of this to the Engine cloud service; the responses from your GraphQL service stay on-prem.
+
+If you've configured Engine caching and Engine determines that a response it sees is cacheable, then the response will be stored in your [Engine cache](./caching.html#config.stores) (either in memory or as an external memcache you configure).
+
+Engine reports the query variables for every trace sample it stores. You can tell Engine to ignore private variables using the [`privateVariables`](./proxy-config.html#Reporting) configuration option in the proxy or you can prevent all variables from being reported with [`noTraceVariables`](./proxy-config.html#Reporting).
```
Comment (stubailo) [inline source/data-privacy.md]: The information about variables is pretty hidden here - maybe we can make a "request" and "response" section? Then the variable and header information can go into the "response" section.

**Comment by justinanastos** [comment]
@stubailo Updated: 

* Update `data` comment
* Better emphasize the variables section

Doesn't address what happens with the rest of the headers, waiting for a response from those in the know.

Also, can I get write access to this repository @daniman or @stubailo ?

**Comment by justinanastos** [comment]
@stubailo All feedback addressed!

**Comment by stubailo** [comment]
@justinanastos we usually make sure to explicitly set IDs in the headers, so that we can change the header text without messing up the anchor links. Made that change for you, otherwise this is awesome!

**Pull request merged**